ui/job-view/headerbars/WatchedRepo.jsx

```diff
@@ -102,12 +102,13 @@ class WatchedRepo extends React.Component {
         messageOfTheDay: '',
         statusInfo: statusInfoMap.unsupported,
       });
+      setCurrentRepoTreeStatus('unsupported');
       clearInterval(this.treeStatusIntervalId);
     } else {
       TreeStatusModel.get(watchedRepoName).then((data) => {
         const treeStatus = data.result;
 
-        if (watchedRepoName === repoName) {
+        if (mustBeCurrentRepo || watchedRepoName === repoName) {
```

Comment:
I'm not sure I understand the need for this addition. If `mustBeCurrentRepo` is `true`, I presume the existing comparison will also be `true`. If it's `false`, the existing comparsion would also be `false`?

Reply:
Unfortunately, that's not the case. If one switches to a different tree, `watchedRepoName` is the new one and `repoName` the old one. Adding the update to the callback of `setState` gets rejected because async code in component mounting is forbidden according to console output.

Reply:
Ah, thanks for explaining that!